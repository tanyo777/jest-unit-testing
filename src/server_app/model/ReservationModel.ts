export type Reservation = {
  id: string | undefined;
  room: string | undefined;
  user: string | undefined;
  startDate: string | undefined;
  endDate: string | undefined;
};
