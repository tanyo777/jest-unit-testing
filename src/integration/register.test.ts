import supertest from 'supertest';
import { HTTP_CODES } from '../server_app/model/ServerModel';
import { Server } from '../server_app/server/Server';

describe('Integration tests for the register endpoint', () => {
  let server: Server;

  beforeAll(() => {
    server = new Server();
    server.startServer();
  });

  afterAll(() => {
    server.stopServer();
  });

  it('should test register endpoint with valid username and password', () => {
    return supertest(server.server)
      .post('/register')
      .send({ userName: 'tanyo', password: 'Test@123' })
      .expect(HTTP_CODES.CREATED)
      .then((response) => {
        const resBody = response.body;
        expect(resBody).toHaveProperty('userId', expect.any(String));
      });
  });

  it('should test register endpoint with empty body', () => {
    return supertest(server.server)
      .post('/register')
      .send({})
      .expect(HTTP_CODES.BAD_REQUEST)
      .then((response) => {
        const resBody = response.body;
        expect(resBody).toBe('userName and password required');
      });
  });

  it('should test the login with correct username and password', () => {
    return supertest(server.server)
      .post('/login')
      .send({ userName: 'tanyo', password: 'Test@123' })
      .expect(HTTP_CODES.CREATED)
      .then((response) => {
        const resBody = response.body;
        expect(resBody).toHaveProperty('token', expect.any(String));
      });
  });

  it('should test the login with incorrect username or password', () => {
    return supertest(server.server)
      .post('/login')
      .send({ userName: 'tany', password: 'Test@12' })
      .expect(HTTP_CODES.NOT_fOUND)
      .then((response) => {
        const resBody = JSON.parse(response.text);
        expect(resBody).toBe('wrong username or password');
      });
  });

  it('should test the login with no username or password field', () => {
    return supertest(server.server)
      .post('/login')
      .send({ password: 'Test@12' })
      .expect(HTTP_CODES.BAD_REQUEST)
      .then((response) => {
        const resBody = JSON.parse(response.text);
        expect(resBody).toBe('userName and password required');
      });
  });

  it('should test the login with ocurred error', () => {
    return supertest(server.server)
      .post('/login')
      .expect(HTTP_CODES.INTERNAL_SERVER_ERROR);
  });
});
