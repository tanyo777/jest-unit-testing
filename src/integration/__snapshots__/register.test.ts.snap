// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration tests for the register endpoint should test the get reservation by id endpoint 1`] = `
{
  "endDate": "testEndDate",
  "id": "2f3b204a3b820f437839",
  "room": "testRoom",
  "startDate": "testStartDate",
  "user": "testUser",
}
`;
